pkg/gardenlet/controller/shoot/care/health.go

```diff
@@ -666,6 +679,32 @@ func (h *Health) CheckClusterNodes(
 	return nil, nil
 }
 
+// CheckNodeAgentLeases checks if all given nodes have a corresponding lease
+func CheckNodeAgentLeases(nodeList *corev1.NodeList, leaseList *coordinationv1.LeaseList, clock clock.Clock) error {
+	if len(leaseList.Items) == 0 {
+		return fmt.Errorf("no leases")
+	}
+
+	leases := map[string]coordinationv1.Lease{}
+	for _, lease := range leaseList.Items {
+		nodeName := strings.ReplaceAll(lease.Name, gardenerutils.NodeLeasePrefix, "")
+		leases[nodeName] = lease
+	}
+
```

Comment:
```suggestion
	// TODO(rfranzke): Remove this if-condition as soon as the UseGardenerNodeAgent feature gate gets removed.
	if len(nodeNameToLease) == 0 {
		return nil // node-agent might not yet be deployed even though the feature gate is turned on, so let's accept this and don't report an error
	}

```

Reply:
Sorry that I misunderstood the comment earlier 😅